### PR TITLE
Change travis to gha

### PR DESCRIPTION
**Pull request by Cielquan**
Like requested in [#33 (comment)](https://github.com/myint/eradicate/pull/33#issuecomment-1085053457) here is a translation of the travis config in GHA.

I updated the tested python versions with additional ones and kept the rest as it was.

EDIT:
I do not intend to rewrite the config to a proper GHA config. Reason see: https://github.com/myint/eradicate/pull/33#issuecomment-1086034653

### CODE REVIEW

**Comment by Cielquan** [comment]
Removed py3.4 - GHA does not support it anymore
Removed py3.5 - `ModuleNotFoundError` used in test_eradicate.py was added in py3.6

**Comment by Cielquan** [comment]
I removed `coveralls` because:

1. the old version errored: https://github.com/Cielquan/eradicate/runs/5790972462?check_suite_focus=true
2. the official coveralls GHA action errors too: https://github.com/Cielquan/eradicate/runs/5791028659?check_suite_focus=true
3. There is no badge or similar on the README indicating the coverage

**Comment by Cielquan** [comment]
Request review by @sobolevn

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["2.7", "3.6", "3.7", "3.8", "3.9", "3.10", "pypy-2.7", "pypy-3.7", "pypy-3.8"]
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v2
```
Comment (sobolevn) [inline .github/workflows/test.yml:24]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+---
+name: Test
+
+on:
+  push:
+    branches:
+      - "**"
```
Comment (sobolevn) [inline .github/workflows/test.yml:7]:
```suggestion
      - master
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  pull_request:
+    branches:
+      - "**"
```
Comment (sobolevn) [inline .github/workflows/test.yml:10]: `pull_request:` is enough.
Please, don't forget `workflow_dispatch`

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+jobs:
+  tests:
+    name: Tests with python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["2.7", "3.6", "3.7", "3.8", "3.9", "3.10", "pypy-2.7", "pypy-3.7", "pypy-3.8"]
+    defaults:
```
Comment (sobolevn) [inline .github/workflows/test.yml:20]: We don't need `bash`, let's use a default.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Install mock
+        run: pip install "mock; python_version>='2.7' and python_version<'2.8'"
+
+      - name: Run test file
+        run: python test_eradicate.py
+
+
+  coverage:
+    name: Run tests with coverage
+    needs: [tests]
```
Comment (sobolevn) [inline .github/workflows/test.yml:47]: I think it can run in parallel.

**Review by sobolevn — COMMENTED** [review]
Thank you!

**Comment by Cielquan** [comment]
Done

**Comment by sobolevn** [comment]
I am going to merge this and fix any problems afterwards.
Otherwise, we won't be able to test this.

Thanks a lot for your help! 👍

**Review by sobolevn — APPROVED** [review]

**Comment by Cielquan** [comment]
There should be none. The workflow ran on my fork and had no issues at the end.

**Pull request merged**